crates/unified-bridge/src/network_id.rs

```diff
@@ -0,0 +1,54 @@
+use std::{fmt::Display, ops::Deref};
+
+use serde::{Deserialize, Serialize};
+
+/// A network ID.
+///
+/// 0 refers to ethereum, and rollups are numbered from 1 to `u32::MAX`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Hash)]
+#[cfg_attr(feature = "testutils", derive(arbitrary::Arbitrary))]
+pub struct NetworkId(u32);
+
+impl Display for NetworkId {
+    #[inline]
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
```

Comment:
Nit: I think this will not preserve various formatting modifiers like `"{my_network_id:5}"`.

For delegating to a member, I generally prefer calling fmt directly:

```suggestion
        self.0.fmt(f)
```

Same applies to `RollupId` and `RollupIndex`.